src/function.ts

```diff
@@ -1,43 +1,94 @@
+import {
+  AssetHashType,
+  aws_lambda,
+  DockerImage,
+  Tokenization,
+} from "aws-cdk-lib";
 import * as appsync from "@aws-cdk/aws-appsync-alpha";
-import { aws_lambda } from "aws-cdk-lib";
 import { CallExpr, isVariableReference } from "./expression";
 import { ASL } from "./asl";
 
 // @ts-ignore - imported for typedoc
 import type { AppsyncResolver, AppSyncVtlIntegration } from "./appsync";
-import { Integration } from "./integration";
+import { NativeFunctionDecl, isNativeFunctionDecl } from "./declaration";
+import { Construct } from "constructs";
+import { AnyFunction } from "./util";
+import { runtime } from "@pulumi/pulumi";
+import path from "path";
+import fs from "fs";
+import { Err, isErr } from "./error";
+import { IntegrationImpl, Integration } from "./integration";
+import { Lambda, EventBridge, StepFunctions } from "aws-sdk";
 
-export function isFunction<P = any, O = any>(a: any): a is Function<P, O> {
+export function isFunction<P = any, O = any>(a: any): a is IFunction<P, O> {
   return a?.kind === "Function";
 }
 
 export type AnyLambda = Function<any, any>;
 
-/**
- * Wraps an {@link aws_lambda.Function} with a type-safe interface that can be
- * called from within an {@link AppsyncResolver}.
- *
- * For example:
- * ```ts
- * const getPerson = new Function<string, Person>(
- *   new aws_lambda.Function(..)
- * );
- *
- * new AppsyncResolver(() => {
- *   return getPerson("value");
- * })
- * ```
- */
-export class Function<P, O> implements Integration {
+export type FunctionClosure<P, O> = (payload: P) => Promise<O>;
```

Comment:
We should pass the Lambda Context from `@types/aws-lambda` as the second argument